src/reduxTypes.js

```diff
@@ -295,6 +295,7 @@ export type SettingsState = {|
   experimentalFeaturesEnabled: boolean,
   streamNotification: boolean,
   browser: BrowserPreference,
+  doNotMarkMessagesAsRead: boolean,
```

Comment:
We're making a change to the Redux state type, so we should have a migration for it.

(More specifically, it's in a subtree that we actually store between runs -- see `storeKeys` and `cacheKeys` in `src/boot/store.js`. This didn't apply to the change to `SessionState` via changing `Debug`, because `state.session` isn't stored; it's in `discardKeys`.)

Actually, we had a recent discussion where the conclusion was that maybe we shouldn't quite have a migration in this case, because the way the persisted state gets merged with the reducer's initial state will have the right effect already. But still we should have a comment in the list of migrations, saying what changed and that we explicitly decided no migration was needed.

(I'm regretting now that that discussion was on GitHub instead of Zulip, though -- it may not be easy to find.)

Reply:
https://github.com/zulip/zulip-mobile/pull/4798#discussion_r652281059 🙂

Reply:
is there a difference when using `dropCache` vs returning a modified state like some migrations are doing? example - '28'

Reply:
https://chat.zulip.org/#narrow/stream/243-mobile-team/topic/data.20migrations/near/1230029 (same link as below, but here too for cross-reference)